content/en/docs/chart_template_guide/looup_function.md

```diff
@@ -0,0 +1,50 @@
+---
+title: "Lookup Function"
+description: "Looking values from resources in a running cluster"
+weight: 17
+---
+
+The `lookup` can be used to reources from resources in a running cluster.
+The sinopsys of the lookup function is:
+`lookup apiVersion(string, not-empty), kind (string, not-empty), namespace(string, can be empty), name(string, can be empty) -> resource map[string]interface{}`
+
+Name and namespace are optional and can be passed an empty string `""`. The following combination of parameters are possible:
+
+| Namespaced Resource | Namespace | Name | Behavior |
+|:---:|:---:|:---:|:---:|
+| yes | non-empty | non-empty | `kubectl get type name -n namespace` |
+| yes | non-empty | empty | `kubectl get type -n namespace`, this returns a list |
+| yes | empty | non-empty | `kubectl get type --all-namespaces` name is ignored, this returns a list |
+| yes | empty | empty | `kubectl get type --all-namespaces`, this returns a list |
+| no | non-empty | non-empty | `kubectl get type name`, namespace is ignored |
+| no | non-empty | empty | `kubectl get type`, namespace is ignored, this returns a list |
+| no | empty | non-empty | `kubectl get type name` |
+| no | empty | empty | `kubectl get type`, this returns a list |
+
+When the function call is designed to return an object, it will return a `map[string]interface{}` obtained from the parsing the json of the returned resource. This structure can the be further navigated to extract specific values, for example:
```

Comment:
`map[string]interface{}` is a Go programming language element. Helm is used by many people who don't know Go so this isn't something they understand. If you look at the sprig docs for [lists](https://masterminds.github.io/sprig/lists.html) and [dictionaries](https://masterminds.github.io/sprig/dicts.html) (more general terms) you'll see descriptions of ideas using more general language. Can we get the language to be more general?

Reply:
fixed see if it's better.